pkg/client/buildconfigs.go

```diff
@@ -130,3 +142,84 @@ func (c *buildConfigs) InstantiateBinary(request *buildapi.BinaryBuildRequestOpt
 		Body(r).Do().Into(result)
 	return
 }
+
+type resourceMapping struct {
+	Kind, Resource string
+	RawJSON        []byte
+	IsOrigin       bool
+}
+
+// templateObjectsMap converts the runtime objects into a map of resources and
+// their JSON representation.
+func objectsToResourceMap(input []runtime.Object) ([]resourceMapping, error) {
+	result := make([]resourceMapping, len(input))
+	var resultErr error
+	for index, item := range input {
+		rawObj, ok := item.(*runtime.Unknown)
+		if !ok {
+			resultErr = fmt.Errorf("unable to convert %+v to unknown object", item)
+			break
+		}
+		obj, err := runtime.Decode(kapi.Codecs.UniversalDecoder(), rawObj.RawJSON)
+		if err != nil {
+			resultErr = fmt.Errorf("unable to decode %+v: %v", rawObj.RawJSON, err)
+			break
+		}
+		kind, err := kapi.Scheme.ObjectKind(obj)
+		if err != nil {
+			resultErr = fmt.Errorf("unable to get kind from %+v", obj)
+			break
+		}
+		plural, _ := meta.KindToResource(kind)
+		result[index] = resourceMapping{
+			Kind:     kind.Kind,
+			Resource: plural.Resource,
+			RawJSON:  rawObj.RawJSON,
+			IsOrigin: latest.IsKindInAnyOriginGroup(kind.Kind),
+		}
+	}
+	return result, resultErr
+}
+
+func (c *buildConfigs) InstantiatePipeline(config *buildapi.JenkinsPipelineBuildStrategy, k kclient.Interface) error {
+	// TODO: Should the template be configured at strategy level?
```

Comment:
This is some sort of helper library that is associated with the controller and the master config, not with clients.